sharktank/sharktank/types/theta.py

```diff
@@ -114,16 +114,18 @@ def transform(self, *transforms: InferenceTensorTransform) -> "Theta":
     def to(self, *, device: Optional[Union[str, torch.device]] = None) -> "Theta":
         return self.transform(InferenceTensorTransforms.to_device(device))
 
-    def pop(self, *name_path: str | int) -> "Theta":
+    def pop(self, *name_path: str | int, inplace: bool = False) -> "Theta":
```

Comment:
I think the `inplace` argument needs some explanation.
It is more natural to have a function with another name as `pop` should mean that you modify the object.
Here we are doing a filtering by subpath, so something like
```
def match(self, *name_path: str | int) -> "Theta":
```